fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+
+The `CreateVersionHistory` class controls the creation the `VERSION_HISTORY` table. Importantly, the VERISON_HISTORY table does not support migrations and changes. The Primary Key has a corresponding index `PK_VERSION_HISTORY`, and the code does not support updating the index. 
+
+This table is an administrative table and should not require updating and migration. 
+
+**Table: TENANTS**
+
+The `FHIR_ADMIN.TENANTS` table maps a given tenant to the Multi-Tenant ID (MT_ID) for the given tenant.  The MT_ID is subsequently used to restrict row access based on the matching value. 
+
+`AdminSchemaGenerator.addTenantTable` method adds the TENANT table, and updates to this table definition requries a change to the setVersion (incrementing by 1). A corresponding change must be made to `FHIRSchemaGenerator.addTenantTable` method.
+
+``` java
+this.tenantsTable = Table.builder(adminSchemaName, TENANTS)
+    .setVersion(1)
+    .addIntColumn(            MT_ID,             false)
+    .addVarcharColumn(      TENANT_NAME,        36,  false)
+    .addVarcharColumn(    TENANT_STATUS,        16,  false)
+    .addUniqueIndex(IDX + "TENANT_TN", TENANT_NAME)
+    .addPrimaryKey("TENANT_PK", MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANTS` and the corresponding indices. 
+
+**Table: TENANT_KEYS**
+
+The Tenant Keys table stores a hashed version of the tenant specific key. Upon creating a session and calling the SET_TENANT procedure, the SET_TENANT queries the incoming key against the TENANT_KEYS database to map to a specific tenant MT_ID. 
+
+`AdminSchemaGenerator.addTenantKeysTable` method adds the TENANT_KEYS table, and updates to this table definition requries a change to the setVersion (incrementing by 1).
+
+``` java
+this.tenantKeysTable = Table.builder(adminSchemaName, TENANT_KEYS)
+    .addIntColumn(        TENANT_KEY_ID,             false)
+    .addIntColumn(                MT_ID,             false)
+    .addVarcharColumn(      TENANT_SALT,        44,  false)
+    .addVarbinaryColumn(    TENANT_HASH,        32,  false)
+    .addUniqueIndex(IDX + "TENANT_KEY_SALT", TENANT_SALT)
+    .addUniqueIndex(IDX + "TENANT_KEY_TIDH", MT_ID, TENANT_HASH)
+    .addPrimaryKey("TENANT_KEY_PK", TENANT_KEY_ID)
+    .addForeignKeyConstraint(FK + TENANT_KEYS + "_TNID", adminSchemaName, TENANTS, MT_ID)
+    .setTablespace(fhirTablespace)
+    .build(model);
+```
+
+If this table is updated, the changes must be manually apply the migration steps to the `TENANT_KEYS` and the corresponding indices. 
```

Comment:
Note sure what this line is trying to say.  Maybe just leave it at "the ADMIN schema is not strictly versioned and so changes to these tables is not supported at this time"?

Reply:
done